tests/oss-fuzz/split_uri_target.c

```diff
@@ -0,0 +1,8 @@
+#include <coap/uri.h>
+
+int
+LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+    coap_uri_t uri;
+    coap_split_uri((unsigned char *)data, size, &uri);
```

Comment:
`coap_split_uri()` takes a `const uint8_t *` as its first argument, so you should remove the typecast.

Reply:
Moreover, you should include `coap/coap.h` instead of `coap/uri.h` as coap.h ensures that libcoap.h is included. Otherwise, definitions are missing:

```
In file included from ../../include/coap/uri.h:15,
                 from split_uri_target.c:1:
../../include/coap/str.h:41:16: error: expected ‘;’ before ‘typedef’
 COAP_DEPRECATED
                ^
                ;
 typedef coap_string_t str;
 ~~~~~~~         
```